R-package/R/xgb.Booster.R

```diff
@@ -1,180 +1,88 @@
-# Construct an internal xgboost Booster and return a handle to it.
+# Construct an internal xgboost Booster and get its current number of rounds.
 # internal utility function
-xgb.Booster.handle <- function(params, cachelist, modelfile, handle) {
+# Note: the number of rounds in the C booster gets reset to zero when updating
+# the parameters through 'xgb.parameters', hence the need to return it from
+# this function when using it for training continuation
+xgb.Booster <- function(params, cachelist, modelfile, training_continuation) {
   if (typeof(cachelist) != "list" ||
       !all(vapply(cachelist, inherits, logical(1), what = 'xgb.DMatrix'))) {
     stop("cachelist must be a list of xgb.DMatrix objects")
   }
   ## Load existing model, dispatch for on disk model file and in memory buffer
   if (!is.null(modelfile)) {
-    if (typeof(modelfile) == "character") {
+    if (is.character(modelfile)) {
       ## A filename
-      handle <- .Call(XGBoosterCreate_R, cachelist)
+      bst <- .Call(XGBoosterCreate_R, cachelist)
       modelfile <- path.expand(modelfile)
-      .Call(XGBoosterLoadModel_R, handle, enc2utf8(modelfile[1]))
-      class(handle) <- "xgb.Booster.handle"
+      .Call(XGBoosterLoadModel_R, xgb.get.handle(bst), enc2utf8(modelfile[1]))
+      niter <- xgb.nrounds(bst)
       if (length(params) > 0) {
-        xgb.parameters(handle) <- params
+        xgb.parameters(bst) <- params
       }
-      return(handle)
-    } else if (typeof(modelfile) == "raw") {
+      return(list(bst = bst, niter = niter))
+    } else if (is.raw(modelfile)) {
       ## A memory buffer
-      bst <- xgb.unserialize(modelfile, handle)
+      bst <- xgb.unserialize(modelfile)
```

Comment:
Can we use the function `xgb.load.raw`, which corresponds to the `xgb.save.raw` function?

In general, we have three sets of IO procedures:
- Save the model to a file, without hyper-parameters like the number of threads.
```
XGBoosterSaveModel
XGBoosterLoadModel
```
`xgb.save` is using this one. Preferably we can rename it to `save.model`. No strong opinion there.

- Save the model to a buffer, without hyper-parameters like the number of threads.
```
XGBoosterSaveModelToBuffer
XGBoosterLoadModelFromBuffer
```
`xgb.save.raw` is using this one.
- Save everything to a buffer, including hyper-parameters.
```
XGBoosterSerializeToBuffer
XGBoosterUnserializeFromBuffer
```
`xgb.serialize` is using this one.

We need to make sure the right function is called.

Reply:
Changed to `xgb.load.raw`.